lib/staticcacher.js

```diff
@@ -11,9 +9,43 @@ function StaticCacher(fileList) {
 
 StaticCacher.prototype.onInstall = function sc_onInstall() {
   var self = this;
-  return CacheHelper.getDefaultCache().then(function(cache) {
-    return CacheHelper.addAll(cache, self.files);
+  return this.getDefaultCache().then(function(cache) {
+    return self.addAll(cache, self.files);
+  });
+};
+
+StaticCacher.prototype.getDefaultCache = function sc_getDefaultCache() {
+  return caches.open('offline');
+};
+
+StaticCacher.prototype.addAll = function(cache, urls) {
+  if (!cache) {
+    throw new Error('Need a cache to store things');
+  }
+  // Polyfill until chrome implements it
+  if (typeof cache.addAll !== 'undefined') {
+    return cache.addAll(urls);
+  }
+
+  var promises = [];
+  var self = this;
+  urls.forEach(function(url) {
+    promises.push(self.fetchAndCache(new Request(url), cache));
+  });
+
+  return Promise.all(promises);
+};
+
+StaticCacher.prototype.fetchAndCache = function sc_fetchAndCache(request, cache) {
+
+  return fetch(request.clone()).then(function(response) {
+    var clone = response.clone();
```

Comment:
That was some legacy from [sw-cache-helper](https://github.com/arcturus/sw-cache-helper/blob/f255110c36f785051c6b96b6534db2bfad54cd24/lib/cachehelper.js#L15). But okay, I changed it.